### PR TITLE
Convert lettuce 5.1 reactive client tests from groovy to java

### PR DESCRIPTION
**Pull request by jaydeluca**
Related to #7195

### CODE REVIEW

File: instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+                            .hasEventsSatisfyingExactly(
+                                event -> event.hasName("redis.encode.start"),
+                                event -> event.hasName("redis.encode.end"))));
+  }
+
+  // to make sure instrumentation's chained completion stages won't interfere with user's, while
+  // still recording spans
+  @Test
+  void testGetNonExistentKeyCommand()
+      throws ExecutionException, InterruptedException, TimeoutException {
```
Comment (breedx-splk) [inline instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java:149]: I think it's perfectly reasonable to just `throws Exception` on all these. Reads a bit clearer to me.

File: instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_ADDR, "127.0.0.1"),
+                                equalTo(
+                                    SemanticAttributes.NET_SOCK_PEER_NAME,
+                                    expectedHostAttributeValue),
+                                equalTo(SemanticAttributes.NET_SOCK_PEER_PORT, port),
+                                equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                                equalTo(SemanticAttributes.DB_STATEMENT, "GET NON_EXISTENT_KEY"))
+                            .hasEventsSatisfyingExactly(
+                                event -> event.hasName("redis.encode.start"),
+                                event -> event.hasName("redis.encode.end")),
```
Comment (breedx-splk) [inline instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java:191]: Missing the `.hasParent(trace.getSpan(0))` for `span(1)` here (it was in the original).

**Review by breedx-splk — APPROVED** [review]
Looks like a pretty clean port to me, thanks! I just had 2 small comments, otherwise looks good. Appreciate it!

File: instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java (hunk 1 above)
Comment (jaydeluca) [reply instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java:149]: I tried updating these but encountered an issue:

```
warning: [InterruptedExceptionSwallowed] This method can throw InterruptedException but declares that it throws Exception/Throwable. This makes it difficult for callers to recognize the need to handle interruption properly.
  void testSetCommandWithSubscribeOnDefinedConsumer()
       ^
    (see https://errorprone.info/bugpattern/InterruptedExceptionSwallowed)
  Did you mean 'throws ExecutionException, InterruptedException, TimeoutException {'?
```

Comment (jaydeluca) [reply instrumentation/lettuce/lettuce-5.1/testing/src/main/java/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceReactiveClientTest.java:150]: nvm, i'll just suppress that

**Pull request merged**